lib/srv/forward/sshserver.go

```diff
@@ -893,6 +893,25 @@ func (s *Server) handleSessionChannel(ctx context.Context, nch ssh.NewChannel) {
 func (s *Server) dispatch(ctx context.Context, ch ssh.Channel, req *ssh.Request, scx *srv.ServerContext) error {
 	scx.Debugf("Handling request %v, want reply %v.", req.Type, req.WantReply)
 
+	if scx.JoinOnly {
+		switch req.Type {
+		case sshutils.PTYRequest:
+			return s.termHandlers.HandlePTYReq(ch, req, scx)
+		case sshutils.ShellRequest:
+			return s.termHandlers.HandleShell(ch, req, scx)
+		case sshutils.WindowChangeRequest:
+			return s.termHandlers.HandleWinChange(ch, req, scx)
+		case teleport.ForceTerminateRequest:
+			return s.termHandlers.HandleForceTerminate(ch, req, scx)
+		case sshutils.EnvRequest:
+			return s.handleEnv(ch, req, scx)
+		case sshutils.SubsystemRequest:
+			return s.handleSubsystem(ch, req, scx)
```

Comment:
Does a join-only user need setenv or subsystems? Same question applies to `lib/srv/regular/sshserver.go`.

Reply:
They do need subsystem but we can get rid of setenv, done.